pkg/handler/config.go

```diff
@@ -54,10 +54,11 @@ type Config struct {
 	// PreUploadCreateCallback will be invoked before a new upload is created, if the
 	// property is supplied. If the callback returns no error, the upload will be created
 	// and optional values from HTTPResponse will be contained in the HTTP response.
+	// Furthermore, updated metadata can be returned by the hook.
 	// If the error is non-nil, the upload will not be created. This can be used to implement
 	// validation of upload metadata etc. Furthermore, HTTPResponse will be ignored and
 	// the error value can contain values for the HTTP response.
-	PreUploadCreateCallback func(hook HookEvent) (HTTPResponse, error)
+	PreUploadCreateCallback func(hook HookEvent) (HTTPResponse, map[string]string, error)
```

Comment:
I am currently thinking about how we can keep this maintainable in the future. Your PR is interested in modifying the meta data but in the near future we also want to modify the upload ID, storage location etc. I think we should add some struct to hold these potential values here. What do you think?

Reply:
an alternative implementation, i wrote here #916 - that should solve ID and Metadata

Reply:
Hi @Acconut 
in https://github.com/tus/tusd/pull/916 you wrote you "would like to avoid merging PR". 
Does this also relate to this pull request? Or should we implement your suggested change and add some struct to hold the return values?

Reply:
Frankly, I am not entirely sure about #916. Through this pointer, it allows the entire struct to be modified, but some of these properties must not be modified by the hook (i.e. Offset, IsFinal, IsPartial). We could just state this rule in the documentation, but it would also be nice if we can enforce it in the API itself.

> in https://github.com/tus/tusd/pull/916 you wrote you "would like to avoid merging PR".

Let me rephrase. I do not want to merge a PR, which addresses some part of the problem of modifying the upload through the pre-create hook, without having the bigger picture on how we can provide this feature with all of its aspects. So if we agree on a way that any kind of hooks can modify relevant properties in a safe manner, I am happy to merge PRs implementing this.

Reply:
Hi @Acconut 
I adjusted the pull request. Please take a look at it and check it out.